src/node/miner.cpp

```diff
@@ -191,33 +192,66 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
 
     // Create coinbase transaction.
     CMutableTransaction coinbaseTx;
-    coinbaseTx.vin.resize(1);
-    coinbaseTx.vin[0].prevout.SetNull();
-    coinbaseTx.vout.resize(1);
-    coinbaseTx.vout[0].scriptPubKey = scriptPubKeyIn;
-    coinbaseTx.vout[0].nAsset = policyAsset;
-    coinbaseTx.vout[0].nValue = nFees + GetBlockSubsidy(nHeight, chainparams.GetConsensus());
-    if (g_con_elementsmode) {
-        if(chainparams.GetConsensus().subsidy_asset != policyAsset) {
-            // Only claim the subsidy if it's the same as the policy asset.
-            coinbaseTx.vout[0].nValue = nFees;
+    if (feeMap.size() > 0) {
+        coinbaseTx.vin.resize(feeMap.size());
```

Comment:
This should be pulled out into a separate function. It's more or less identical to the `else` block except that it iterates over the fee map to create a single output per asset.